### PR TITLE
Display Class Finder entries on separate lines

### PR DESCRIPTION
**Pull request by jkmartindale**
Cosmetic fix; `static-menu.pt` already does this, but the change was never carried
over to `menu.pt`

### CODE REVIEW

**Comment by jamadden** [comment]
LGTM. Any thoughts @mgedmin ?

File: src/zope/app/apidoc/codemodule/browser/menu.pt [hunk 1]
```diff
@@ -22,13 +22,13 @@
 
     <div tal:define="classes view/findClasses"
          tal:condition="classes">
-
-      <a href="" target="main"
-         tal:repeat="info classes"
-         tal:attributes="href info/url"
-         tal:content="info/path">
+      <div tal:repeat="info classes" style="display:block">
```
Comment (mgedmin) [inline src/zope/app/apidoc/codemodule/browser/menu.pt]: Is there some CSS that changes the presentation of divs that requires an explicit style="display: block" here?

Comment (jkmartindale) [reply src/zope/app/apidoc/codemodule/browser/menu.pt]: There is no such style rule, and I think using line breaks between links makes much more sense anyway.

**Comment by jkmartindale** [comment]
Oops, that's not how TAL works.

File: src/zope/app/apidoc/codemodule/browser/menu.pt (hunk 1 above)
Comment (mgedmin) [reply src/zope/app/apidoc/codemodule/browser/menu.pt]: Huh?  `<div>` elements are block-level elements by default, so you should be able to drop the `style` attribute and get the desired effect anyway?  I feel like I'm misunderstanding something here.

File: src/zope/app/apidoc/codemodule/browser/static_menu.pt [hunk 2]
```diff
@@ -7,44 +7,37 @@
     <div>
       <span i18n:translate="">Class Finder:</span>
       <span><i i18n:translate="">(Enter partial Python path)</i></span>
-	</div>
+	  </div>
```
Comment (mgedmin) [inline src/zope/app/apidoc/codemodule/browser/static_menu.pt]: The indentation here doesn't match the opening `</div>`.

I get the feeling your editor tried to be helpful, but made things worse?

File: src/zope/app/apidoc/codemodule/browser/menu.pt [hunk 3]
```diff
@@ -28,10 +29,10 @@
          tal:attributes="href info/url"
          tal:content="info/path">
         /zope/app/Application
-      </a>
+      </a><br />
```
Comment (mgedmin) [inline src/zope/app/apidoc/codemodule/browser/menu.pt]: You're adding one `<br />` after all the repeated `<a>` and I don't think that was your intention.

File: src/zope/app/apidoc/codemodule/browser/static_menu.pt [hunk 4]
```diff
@@ -7,44 +7,37 @@
...
-      <div tal:repeat="info pyclasses" style="display:block"
-           tal:attributes="id info/counter">
-        <a href="" target="main"
-           tal:attributes="href info/url"
-           tal:content="info/path">
-         /zope/app/Application
-        </a>
-      </div>
+    <div tal:define="classes view/findClasses"
+         tal:condition="classes">
```
Comment (mgedmin) [inline src/zope/app/apidoc/codemodule/browser/static_menu.pt]: I don't quite understand why you're dropping `id` attributes here, which may be a sign that I'm a bad person to review zope.app.apidoc's PRs.

(I like documentation generators, I just don't actually use zope.app.apidoc today ... or at any point in the last ten years, I think.)

File: src/zope/app/apidoc/codemodule/browser/static_menu.pt (hunk 2 above)
Comment (jkmartindale) [reply src/zope/app/apidoc/codemodule/browser/static_menu.pt]: Yeah, I uh, managed to really mess up a lot of things at once here.

File: src/zope/app/apidoc/codemodule/browser/static_menu.pt (hunk 4 above)
Comment (jkmartindale) [reply src/zope/app/apidoc/codemodule/browser/static_menu.pt]: I copied and pasted without checking the differences between `menu.pt` and `static_menu.pt`. I haven't checked to see if the counter IDs are necessary, but the `resultlist` ID is definitely necessary for the JavaScript search function. I also didn't realize that replacing `static_menu.pt`'s `view/findAllClasses` for `menu.pt`'s `view/findClasses` would break the class listing.  
  
All in all, you're a much better reviewer than I am a committer.

Comment (mgedmin) [reply src/zope/app/apidoc/codemodule/browser/static_menu.pt]: I didn't even notice the `findClasses` vs `findAllClasses` difference!

**Comment by mgedmin** [comment]
You can always force-push the branch after cleaning it up (e.g. with `git rebase -i`), and GitHub will update the pull request.

**Comment by jkmartindale** [comment]
My issue is I meant to push to my fork, but instead committed to the upstream Zope Foundation repository.

**Comment by jamadden** [comment]
Yup, I see unwanted commits on `master` here. I'll back them out.

**Comment by jamadden** [comment]
https://github.com/zopefoundation/zope.app.apidoc/commit/45786f1b802e840c54cf8c3a45220b7d9fc0f538 was removed from master.

**Comment by jkmartindale** [comment]
I think everything is finalized. Does this warrant an addition to `CHANGES.rst`?

File: src/zope/app/apidoc/codemodule/browser/menu.pt [hunk 5]
```diff
@@ -22,13 +23,13 @@
...
     <div tal:define="classes view/findClasses"
          tal:condition="classes">
-
-      <a href="" target="main"
-         tal:repeat="info classes"
-         tal:attributes="href info/url"
-         tal:content="info/path">
-        /zope/app/Application
-      </a>
+      <tal:omit-tag tal:repeat="info classes">
```
Comment (jamadden) [inline src/zope/app/apidoc/codemodule/browser/menu.pt]: Why was the 'tal:repeat' moved? Can it go back the way it was?

File: src/zope/app/apidoc/codemodule/browser/static_menu.pt [hunk 6]
```diff
@@ -1,48 +1,39 @@
...
-      <span i18n:translate="">Class Finder:</span>
-      <span><i i18n:translate="">(Enter partial Python path)</i></span>
-	</div>
-    <form action="#" method="post" name="searchform" >
-	  <br />
+      <span i18n:translate="">Class Finder:</span><br />
+      <em i18n:translate="">(Enter partial Python path)</em>
+    </div>
+    <form action="" method="post" name="searchform"
+          onsubmit="simplegetSearchResult(document.searchform.path.value);
```
Comment (jamadden) [inline src/zope/app/apidoc/codemodule/browser/static_menu.pt]: The changes to this file seem unrelated to the purpose of this PR. Can they be separated? 

(Actually I don't even understand why they would be necessary, this just seems to change one type of button to another. Then again, I haven't thought at this level in awhile.)

File: src/zope/app/apidoc/codemodule/browser/static_menu.pt [hunk 7]
```diff
@@ -1,48 +1,39 @@
...
     </p>
 
-	<div>
+    <div>
       <span i18n:translate="">Search results:</span><br /><br />
     </div>
 
-    <div tal:define="pyclasses view/findAllClasses"
-         tal:condition="pyclasses" id="resultlist">
+    <div tal:define="classes view/findAllClasses"
```
Comment (jamadden) [inline src/zope/app/apidoc/codemodule/browser/static_menu.pt]: I don't understand why it was necessary to change this variable name?

File: src/zope/app/apidoc/codemodule/browser/menu.pt (hunk 5 above)
Comment (jkmartindale) [reply src/zope/app/apidoc/codemodule/browser/menu.pt]: Because link elements are inline, the class finder links don't each get their own line without additional help. This leads to some short paths sharing a single line (especially if the side frame is expanded) which can be confusing.
image
 `static_menu.pt` fixes this by wrapping the repeated link in a div containing the `tal:repeat` attribute. For `menu.pt` I didn't see the point in wrapping the links in a lot of useless divs like in its static counterpart (which uses the divs for the search function as well), so I used line break tags instead. Either way, the `tal:repeat` attribute needs to be moved to a block container around the link in order to make them show up on individual lines.
image

File: src/zope/app/apidoc/codemodule/browser/static_menu.pt (hunk 7 above)
Comment (jkmartindale) [reply src/zope/app/apidoc/codemodule/browser/static_menu.pt]: It wasn't

File: src/zope/app/apidoc/codemodule/browser/menu.pt (hunk 5 above)
Comment (jamadden) [reply src/zope/app/apidoc/codemodule/browser/menu.pt]: D'oh, of course. I completely spaced on the `<br />` tag hiding down there. Maybe it should be on its own line so it's more visible.

Comment (jkmartindale) [reply src/zope/app/apidoc/codemodule/browser/menu.pt]: Alright, I moved it and updated `CHANGES.rst`.
